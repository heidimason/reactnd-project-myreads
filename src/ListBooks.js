import React, { Component } from 'react'
import ShelfChanger from './ShelfChanger'

class ListBooks extends Component {
	render() {
        const { bookshelf, showingBooks, onMoveBook } = this.props

        console.log(bookshelf)

        return (
            <div>
                <div className="bookshelf-books">
                    <ol className="books-grid">
						{showingBooks.map( (book, index) => (
          					<li key={index}>
            					<div className="book">
              						<div className="book-top">
                                        {/* If there is a book thumbnail */}
                                        {book.imageLinks &&
                    						<div className="book-cover"
                                                style={{ backgroundImage: 'url(' + book.imageLinks.thumbnail + ')' }}
                                                alt={book.title}>
                                            </div>
                                        }

                                        {/* If there is no book thumbnail */}
                                        {!book.imageLinks &&
                                            <div className="book-cover"
                                                alt={book.title}>
                                            </div>
                                        }

                						<ShelfChanger
                                            showingBooks={showingBooks}
                                            bookshelf={bookshelf}
                                            onMoveBook={ (newShelf) => onMoveBook(book, newShelf)}
                                        />
              						</div>

              						<div className="book-title">{book.title}</div>

                                    {/* If there is a book author */}
                                    {book.imageLinks &&
          						        <div className="book-authors">{book.authors}</div>
                                    }
            					</div>
          					</li>
                        ))}
                    </ol>
                </div>
            </div>
		)
	}
}

export default ListBooks