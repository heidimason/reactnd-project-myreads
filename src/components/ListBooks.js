import React from 'react'
import ShelfChanger from './ShelfChanger'

const ListBooks = props => {
    const { showingBooks, onMoveBook } = props

    return (
        <div>
            <div className="bookshelf-books">
                <ol className="books-grid">
					{showingBooks.map( (book, index) => (
      					<li key={index}>
        					<div className="book">
          						<div className="book-top">
            						<div className="book-cover"
                                        style={{ backgroundImage: `url(${book.imageLinks && book.imageLinks.thumbnail ? `${book.imageLinks.thumbnail}` : `http://via.placeholder.com/128x193?text=No%20Cover`})` }}
                                        alt={book.title}>
                                    </div>


            						<ShelfChanger
                                        bookshelf={book.shelf}
                                        showingBooks={showingBooks}
                                        onMoveBook={ shelf => onMoveBook(book, shelf) }
                                    />
          						</div>

          						<div className="book-title">{book.title}</div>

                                { /* If there is a book author */ }
                                {book.authors &&
      						        <div className="book-authors">{book.authors.map( author => author + '\n' )}</div>
                                }
        					</div>
      					</li>
                    ))}
                </ol>
            </div>
        </div>
	)
}

export default ListBooks