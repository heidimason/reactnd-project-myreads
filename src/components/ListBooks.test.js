import React from 'react'
import renderer from 'react-test-renderer'
import ListBooks from './ListBooks'

describe('<ListBooks />', () => {
	it('renders correctly when there are no items', () => {
		const tree = renderer.create(
			<ListBooks showingBooks={[]} />
		).toJSON()

		expect(tree).toMatchSnapshot()
	})
})
