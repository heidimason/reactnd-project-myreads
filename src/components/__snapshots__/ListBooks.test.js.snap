exports[`<ListBooks /> renders correctly when there are no items 1`] = `
<div
  className="bookshelf-books">
  <ol
    className="books-grid" />
</div>
`;
